packages/quantic/force-app/custom-token/main/default/classes/CustomTokenProvider.cls

```diff
@@ -0,0 +1,119 @@
+        return (String) responseAsMap.get('token');
+    }
+
+    /**
+     * Build a HttpRequest to get a search token from a TokenSettings__c object.
+     * @param {TokenSettings__c} settings the settings to use.
+     */
+    private static HttpRequest buildHttpRequest(TokenSettings__c settings) {
+        HttpRequest tokenRequest = new HttpRequest();
+        tokenRequest.setEndpoint(settings.PlatformUrl__c + TOKEN_PATH + '?organizationId=' + settings.OrganizationId__c);
+        tokenRequest.setMethod('POST');
+        tokenRequest.setHeader('Accept', CONTENT_TYPE_APPLICATION_JSON);
+        tokenRequest.setHeader('Content-Type', CONTENT_TYPE_APPLICATION_JSON);
+        tokenRequest.setHeader('Authorization', 'Bearer ' + settings.ApiKey__c);
+
+        Map<String,Object> requestBody = new Map<String, Object>();
+        requestBody.put('userIds', CustomTokenProvider.getUserIdentities());
+        String requestBodySerialized = JSON.serialize(requestBody);
+        tokenRequest.setBody(requestBodySerialized);
+        tokenRequest.setHeader('Content-Length', String.valueOf(requestBodySerialized.length()));
+
+        return tokenRequest;
+    }
+
+    /**
+     * Get a search token from Coveo Platform based on the setting and the current user identity.
+     * @param {TokenSettings__c} settings the settings to use.
+     */
+    private static String getToken(TokenSettings__c settings) {
+        HttpRequest tokenRequest = CustomTokenProvider.buildHttpRequest(settings);
+
+        HTTPResponse tokenResponse = (new Http()).send(tokenRequest);
+        if(!CustomTokenProvider.isASuccessResponse(tokenResponse)) {
+            throw new HttpException(tokenResponse.getBody());
+        }
+
+        return CustomTokenProvider.extractTokenFromResponse(tokenResponse);
+    }
+
+    /**
+     * Get a serialized object representing a HeadlessConfiguration, based on the TokenSettings__c
+     * of the organization and the user identity of the user that called the method/loaded the page.
+     * @see `ITokenProvider.getHeadlessConfiguration` for more details.
+     */
+    @AuraEnabled(continuation=true cacheable=false)
+    global static String getHeadlessConfiguration() {
+        // Fetch the default value of the TokenSettings__c see https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_custom_settings.htm#apex_System_HierarchyCustomSetting_getOrgDefaults.
+        TokenSettings__c settings = TokenSettings__c.getOrgDefaults();
+
+        if (CustomTokenProvider.isTokenSettingInvalid(settings)) {
+            throw new InvalidTokenSettingsException('TokenSettings invalid. Defaulting to sample organization settings.');
+        }
+
+        Map<String, String> headlessConfiguration = new Map<String, String>();
+        headlessConfiguration.put('organizationId', settings.OrganizationId__c);
+        headlessConfiguration.put('accessToken', CustomTokenProvider.getToken(settings););
```

Comment:
Remove the odd `;` at the end of `.getToken(settings)`. The compiler didn't raise an error for that?